java/ql/src/semmle/code/java/Annotation.qll

```diff
@@ -44,12 +44,65 @@ class Annotation extends @annotation, Expr {
     result = this.getType().getAnnotationElement(name)
   }
 
-  /** Gets a value of an annotation element. */
+  /**
+   * Gets a value of an annotation element. This includes default values in case
+   * no explicit value is specified.
+   */
   Expr getAValue() { filteredAnnotValue(this, _, result) }
 
-  /** Gets the value of the annotation element with the specified `name`. */
+  /**
+   * Gets the value of the annotation element with the specified `name`.
+   * This includes default values in case no explicit value is specified.
+   */
   Expr getValue(string name) { filteredAnnotValue(this, this.getAnnotationElement(name), result) }
 
+  /**
+   * If the value type of the annotation element with the specified `name` is an enum type,
+   * gets the enum constant used as value for that element. This includes default values in
+   * case no explicit value is specified.
+   */
+  EnumConstant getValueEnumConstant(string name) { result = getValue(name).(FieldRead).getField() }
+
+  /**
+   * If the value type of the annotation element with the specified `name` is `String`,
+   * gets the string value used for that element. This includes default values in case no
+   * explicit value is specified.
+   */
+  string getValueString(string name) {
```

Comment:
Rename `getStringValue`, `getIntValue` etc for consistency

Reply:
Your proposed predicate names sound more natural, but I was afraid it would make the CodeQL class more difficult to use. When all predicates for a single value start with `getValue...`, then it is easier to find them and understand their behavior.
When you have `getStringValue`, ... it might be more difficult to tell them apart from `getAValue(name)`.

What do you think?

Reply:
Fair point in general, but I think it's better to stick with what we have than introduce a new convention while keeping the old one elsewhere in the library.

Reply:
Have renamed them to  `get...Value`.